lib/redis/lock.rb

```diff
@@ -38,18 +38,19 @@ def lock(&block)
       try_until_timeout do
         expiration = generate_expiration
         # Use the expiration as the value of the lock.
-        break if redis.setnx(key, expiration)
+        break if redis.setnx(key, expiration) && Redis.next.setnx(key, expiration)
```

Comment:
I can wrap the setting of the key with an if statement that removes the old one if the new one isn't set before breaking. That seems reasonable and would prevent the situation where locks are in the old but not the new.